next-docs/pages/guides/hello-world/add-system.mdx

```diff
@@ -0,0 +1,296 @@
+        import { ClientComponents } from "./createClientComponents";
+        import { SetupNetworkResult } from "./setupNetwork";
+        import { singletonEntity } from "@latticexyz/store-sync/recs";
+
+        export type SystemCalls = ReturnType<typeof createSystemCalls>;
+
+        export function createSystemCalls(
+          /*
+          * The parameter list informs TypeScript that:
+          *
+          * - The first parameter is expected to be a
+          *   SetupNetworkResult, as defined in setupNetwork.ts
+          *
+          * - Out of this parameter, we only care about two fields:
+          *   - worldContract (which comes from createContract, see
+          *     https://github.com/latticexyz/mud/blob/26dabb34321eedff7a43f3fcb46da4f3f5ba3708/templates/vanilla/packages/client/src/mu$
+          *   - waitForTransaction (which comes from syncToRecs, see
+          *     https://github.com/latticexyz/mud/blob/26dabb34321eedff7a43f3fcb46da4f3f5ba3708/templates/vanilla/packages/client/src/mu$
+          *
+          * - From the second parameter, which is a ClientComponent,
+          *   we only care about Counter. This parameter comes to use
+          *   through createClientComponents.ts, but it originates in
+          *   syncToRecs
+        (https://github.com/latticexyz/mud/blob/26dabb34321eedff7a43f3fcb46da4f3f5ba3708/templates/vanilla/packages/client/src/mud/setupN$
+          */
+          { worldContract, waitForTransaction }: SetupNetworkResult,
+          { Counter }: ClientComponents
+        ) {
+          const increment = async () => {
+            /*
+            * Because IncrementSystem
+            * (https://mud.dev/tutorials/walkthrough/minimal-onchain#incrementsystemsol)
+            * is in the root namespace, `.increment` can be called directly
+            * on the World contract.
+            */
+            const tx = await worldContract.write.increment();
+            await waitForTransaction(tx);
+            return getComponentValue(Counter, singletonEntity);
+          };
+
+          const decrement = async () => {
+            const tx = await worldContract.write.decrement();
+            await waitForTransaction(tx);
+            return getComponentValue(Counter, singletonEntity);
+          };
+
+          return {
+            increment,
+            decrement,
+          };
+        }
+        ```
+
+    </CollapseCode>
+
+           <details>
+
+    <summary>Explanation</summary>
+
+        {" "}
+        <p></p>
+
+        The new function is `decrement`.
+
+        ```typescript
+        const decrement = async () => {
+        ```
+
+        This function involves sending a transaction, which is a slow process, so it needs to be [asynchronous](https://www.w3schools.com/js/js_async.asp).
+
+        ```typescript
+          const tx = await worldContract.write.decrement();
+        ```
+
+        This is the way we call functions in top-level systems in a world.
```

Comment:
```suggestion
        This is the way we call functions in systems in a world.
```

I feel like "top-level" is a bit confusing here. We could explain that it's a "root" system and therefore the function selector in the World is the same as the function selector in the system, otherwise it would be `namespace_name_decrement()`, but not sure if this is the right place to explain it. Maybe we should link to another place where it is explained instead.

Reply:
How about "systems in the root namespace"? The place to explain the details is in Extending World, where we will actually do it.

Reply:
yep that works